### PR TITLE
fix: Docker build not reporting errors

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

[This PR](https://github.com/PostHog/posthog/pull/20844/files#diff-0aca4ff4ae1c4070690c9c445d579e54d5435c35fb447c15a1ab34e3a6916a52) (by accident) broke the toolbar bundling but we didn't see it as docker didn't exist with an error

## Changes

* Fixes the esbuild stuff that was silently failing
* Fixes posthog-js imports

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/toolbar/bar/Toolbar.tsx [hunk 1]
```diff
@@ -38,6 +38,8 @@ function MoreMenu(): JSX.Element {
 
     const { logout } = useActions(toolbarConfigLogic)
 
+    // eslint-disable-next-line no-console
+    console.log('testing')
```
Comment (pauldambra) [inline frontend/src/toolbar/bar/Toolbar.tsx:42]:
```suggestion
```

File: production.Dockerfile [hunk 2]
```diff
@@ -23,7 +23,7 @@
 #
 FROM node:18.12.1-bullseye-slim AS frontend-build
 WORKDIR /code
-SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+SHELL ["/bin/bash", "-e", "-o", "pipefail", "-c"]
```
Comment (pauldambra) [inline production.Dockerfile:26]: is this belt & braces or ??

File: frontend/utils.mjs
Comment (pauldambra) [inline frontend/utils.mjs:1]: in dev mode with an error I see

```[ESBUILD] ✘ [ERROR] No matching export in "node_modules/.pnpm/posthog-js-lite@2.5.0/node_modules/posthog-js-lite/lib/index.esm.js" for import "posthog"
[ESBUILD] 
[ESBUILD]     frontend/src/scenes/insights/EmptyStates/EmptyStates.tsx:14:9:
[ESBUILD]       14 │ import { posthog } from 'posthog-js'
[ESBUILD]          ╵          ~~~~~~~
[ESBUILD] 
[ESBUILD] 🥇 "PostHog App": Built in 10.877s
[ESBUILD] 🛑 "Toolbar": Building failed in 10.901s
[TYPEGEN] 🥚 message TS6194: Found 0 errors. Watching for file changes.
```

so it's reporting the failure but continuing to watch 👍

Comment (pauldambra) [reply frontend/utils.mjs:1]: prod build says

```
✘ [ERROR] No matching export in "node_modules/.pnpm/posthog-js-lite@2.5.0/node_modules/posthog-js-lite/lib/index.esm.js" for import "posthog"

    frontend/src/scenes/insights/EmptyStates/EmptyStates.tsx:14:9:
      14 │ import { posthog } from 'posthog-js'
         ╵          ~~~~~~~

 ELIFECYCLE  Command failed with exit code 1.
```

👍

**Review by pauldambra — APPROVED** [review]
still errors in the code but they fail the build

File: production.Dockerfile (hunk 2 above)
Comment (benjackwhite) [reply production.Dockerfile:26]: Was what I originally thought was the issue but it isn't. Nonetheless it doesn't seem to hurt

File: frontend/__snapshots__/scenes-app-insights-error-empty-states--long-loading--light.png
Comment (pauldambra) [inline frontend/__snapshots__/scenes-app-insights-error-empty-states--long-loading--light.png:1]: 🤞 hopefully these are unrelated timing issues

Comment (benjackwhite) [reply frontend/__snapshots__/scenes-app-insights-error-empty-states--long-loading--light.png:1]: Yeah these are in basically every PR right now
